src/features/blocks/Block.tsx

```diff
@@ -1,51 +1,80 @@
 import './Block.css'
 import {ToolboxCategoryBlocks} from "../../toolboxconfig";
 import {useEffect, useRef} from "react";
-import {useAppDispatch} from "../../hooks";
+import {useAppDispatch, useAppSelector} from "../../hooks";
 import {toggleBlockDescriptionAction} from "../../store/features/blocks/blocks";
+import {Draggable} from "react-beautiful-dnd";
+import {getDragStyle} from "../../App";
 
 function Block(props: ToolboxCategoryBlocks) {
+  const cat = useAppSelector(state => state.blocksReducer.categories.find((c) => c.blocks.find((b) => b.id === props.id)))
   const dispatch = useAppDispatch()
   const blockDescriptionRef = useRef<HTMLDivElement>(null)
   const openingTag = '<' + props.tag + '>'
   const closingTag = '</' + props.tag + '>'
 
-  function toggleBlockDescription(block: number) {
-    dispatch(toggleBlockDescriptionAction(block))
-  }
-
+  // TODO Change behavior and fix height inconsistencies
   useEffect(() => {
     (function () {
       let maxHeight = 0
       if (blockDescriptionRef.current && blockDescriptionRef.current.firstElementChild) {
-        if (props.toggled) {
+        if (cat && props.id === cat.openDesc) {
           maxHeight = blockDescriptionRef.current.firstElementChild.getBoundingClientRect().height
         } else {
           maxHeight = 0
         }
         blockDescriptionRef.current.style.maxHeight = maxHeight + "px"
       }
     })()
-  }, [props.toggled])
+  }, [cat, props.id])
+
+  enum TagTypes {
+    Opening = 'opening',
+    Closing = 'closing'
+  }
+
+  function makeToolboxDraggable(tagProp: string, type: TagTypes, index: number) {
+    let classesToAdd = type === 'opening' ? 'toolbox-block-tag tag-open' : 'toolbox-block-tag tag-close'
```

Comment:
TagType.Opening instead of 'opening' :wink:

Reply:
Fixed, thanks!